python/fledge/plugins/south/mqtt-readings/mqtt-readings.py

```diff
@@ -301,10 +309,21 @@ def stop(self):
         self.mqtt_client.disconnect()
         self.mqtt_client.loop_stop()
 
+    def convert(self, msg):
+        constructors = [json.loads, int, float, str]
+        for constructor in constructors:
+            try:
+                # Only convert if type is string
+                converted_msg = constructor(msg) if type(msg) == str else msg
+                # Create dict if converted msg isnt already a dict
+                return converted_msg if type(converted_msg) == dict else {self.json_object_key: converted_msg}
+            except (ValueError, TypeError) as error:
```

Comment:
try except else is more Pythonic way. Also that else with return will make it clear (from readability perspective) that return will ensure if possible type is not in  constructors loop.

Reply:
Thanks for pointing that out, I will try to incorporate it into my future work.